e2e/tests/LoggedIn/crosslinking.e2e.spec.ts

```diff
@@ -0,0 +1,92 @@
+        const page = await Helpers.users.getPageAsUser(browser);
+        // Create new publication
+        await Helpers.publicationCreation.createLivePublication(page, 'Creating a crosslink');
+        // Open suggest crosslink modal
+        await page.locator(PageModel.crosslinks.suggestModal.openButtonLoggedIn).click();
+        await page.waitForSelector(PageModel.crosslinks.suggestModal.searchInput);
+        // Search for existing publication and select it
+        await page.locator(PageModel.crosslinks.suggestModal.searchInput).click();
+        await page.keyboard.type(targetPublicationTitle);
+        await page.locator(`[role="option"]:has-text("${targetPublicationTitle}")`).click();
+        await expect(page.locator(PageModel.crosslinks.suggestModal.clearSelectionButton)).toBeVisible();
+        await expect(page.locator(PageModel.crosslinks.suggestModal.searchInput)).toHaveValue(targetPublicationTitle);
+        // Clear selection
+        await page.locator(PageModel.crosslinks.suggestModal.clearSelectionButton).click();
+        await page.waitForSelector(PageModel.crosslinks.suggestModal.clearSelectionButton, {
+            state: 'hidden'
+        });
+        await expect(page.getByText(targetPublicationTitle)).toHaveCount(0);
+        // Re-select target publication
+        await page.locator(PageModel.crosslinks.suggestModal.searchInput).click();
+        await page.keyboard.type(targetPublicationTitle);
+        await page.locator(`[role="option"]:has-text("${targetPublicationTitle}")`).click();
+        // Save suggestion
+        await Promise.all([
+            page.waitForResponse((response) => response.url().includes('/crosslinks') && response.ok()),
+            page.locator(PageModel.crosslinks.suggestModal.suggestButton).click()
+        ]);
+        await expect(page.getByText('Suggestion created')).toBeVisible();
+        // Check that crosslink is present on new publication's page
+        await page.reload();
+        await expect(
+            page.locator('#desktop-related-publications-items').getByText(targetPublicationTitle)
+        ).toBeVisible();
+    });
+
+    test('Vote on a crosslink', async ({ browser }) => {
+        const page = await Helpers.users.getPageAsUser(browser);
+        await page.goto('/publications/publication-user-5-analysis-1-live');
+
+        // Navigate via crosslink.
+        await page
+            .locator('#desktop-related-publications-items')
+            .getByRole('link', { name: 'View this publication' })
+            .click();
+        await page.waitForURL(
+            '/publications/publication-user-1-problem-1-live/versions/latest?suggestedFrom=publication-user-5-analysis-1-live'
+        );
+
+        const upvoteButton = page.locator(PageModel.crosslinks.votingArea.upvoteButton);
+        const downvoteButton = page.locator(PageModel.crosslinks.votingArea.downvoteButton);
+
+        // If the votes aren't at their initial value, reset them for the test.
```

Comment:
Why wouldn't they be at their initial value? Shouldn't they be at their initial value?

Reply:
I wanted the test to work even if it had not finished gracefully previously and the data was left in an unexpected state.